### ISSUE TITLE
SC2012 seems to be misleading for 'ls -t'

### ISSUE THREAD

**Issue by raxod502** [koalaman/shellcheck#1247]
#### For bugs
- Rule Id: SC2012
- My shellcheck version: 0.4.7
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

The following script opens the most recently modified file in the current directory:

```sh
#!/bin/bash
open "$(ls -t1 | head -n 1)"
```

#### Here's what shellcheck currently says:

```
Line 2:
open "$(ls -t1 | head -n 1)"
        ^-- SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
```

#### Here's what I wanted or expected to see:

No error. As far as I know, there's no way using `find` to accomplish what I'm doing above. Maybe SC2012 should be disabled when `ls` is run with the `-t` flag?

**Comment by asottile** [comment]
```bash
open "$(find . -maxdepth 1 -type f -printf '%T@ %f\n' | sort -r | cut -d' ' -f2- | head -1)"
```

(not perfect, but I don't have to deal with filenames that have newlines in them)

**Comment by koalaman** [comment]
ShellCheck is *technically* correct. You'd optimally want to use a `for` loop that iterates over files to find the latest one (`[[ $current -nt $latest ]]`), or more generally something akin to `find | sort` like @asottile  suggests.

With GNU ls your particular case already handles 99% of cases just fine including spaces and most special characters though. It's also very simple and straight forward. Use your own judgement whether you care enough to introduce the extra complexity to get it to 100% .
